packages/swap/src/event-handlers/swapRequestCompleted.ts

```diff
@@ -35,29 +35,39 @@ export default async function swapRequestCompleted({
   if (
     channelQuote &&
     channelQuote?.expectedDepositAmount.toFixed(0) === swapRequest.depositAmount?.toFixed(0) &&
-    !channelQuote.swapRequestId && // do not overwrite data if channel is reused
-    !swapRequest.refundEgressId // only store execution price if full deposit amount was swapped
+    !channelQuote.swapRequestId // do not overwrite data if channel is reused
   ) {
-    const quotedPrice = channelQuote.quotedPrice.toNumber();
-    const executedPrice = Number(
-      getSwapPrice(
-        swapRequest.srcAsset,
-        swapRequest.swapInputAmount.toFixed(),
-        swapRequest.destAsset,
-        totalSwapOutputAmount.toFixed(),
-      ),
-    );
-    const executedSlippagePercent = (100 * (quotedPrice - executedPrice)) / quotedPrice;
+    if (swapRequest.refundEgressId) {
+      await prisma.quote.update({
+        where: { id: channelQuote.id },
+        data: {
+          refundedAt: new Date(block.timestamp),
+          swapRequestId: swapRequest.id,
+          swapRequestNativeId: swapRequest.nativeId,
+        },
+      });
+    } else {
+      const quotedPrice = channelQuote.quotedPrice.toNumber();
+      const executedPrice = Number(
+        getSwapPrice(
+          swapRequest.srcAsset,
+          swapRequest.swapInputAmount.toFixed(),
+          swapRequest.destAsset,
+          totalSwapOutputAmount.toFixed(),
+        ),
+      );
+      const executedSlippagePercent = (100 * (quotedPrice - executedPrice)) / quotedPrice;
```

Comment:
we could, but i think we dont want to because it would distort the data for analysis in two ways: 
- if a swap partially refunded, its executed slippage was artificially limited (it refunded because the executed slippage would have exceeded the limit). you would need to exclude these swaps to not skew the averages, but thats very easy to forget
- the recommended slippage of the quote was for the quoted amount, but the executed slippage is about the swapped amount. in case of a refund these amounts wont be same, so analyzing the difference between recommended and executed would be misleading